program/rust/src/processor/upd_price.rs

```diff
@@ -171,20 +174,39 @@ pub fn upd_price(
         }
     }
 
+    // TWAP message will be stored here if:
+    // * price_account is V2
+    // * accumulator accounts were passed
+    #[cfg(feature = "pythnet")]
+    let mut maybe_twap_msg = None;
 
-    let account_len = price_account.try_data_len()?;
-    if account_len >= PriceAccountV2::MINIMUM_SIZE {
-        let mut price_data =
+    // Feature-gated PriceAccountV2-specific processing, used only on pythnet/pythtest. May populate
+    // maybe_twap_msg for later use in accumulator message cross-call.
+    //
+    // NOTE: This is done here specifically in order to keep price
+    // account data available for borrowing as v1 for remaining v1 processing.
+    #[cfg(feature = "pythnet")]
+    if price_account.try_data_len()? >= PriceAccountV2::MINIMUM_SIZE {
+        let mut price_data_v2 =
             load_checked::<PriceAccountV2>(price_account, cmd_args.header.version)?;
 
         if aggregate_updated {
-            price_data.update_price_cumulative()?;
+            // Update PriceCumulative on aggregation. This is done
+            // here, because PriceCumulative must update
+            // regardless of accumulator accounts being passed.
+            price_data_v2.update_price_cumulative()?;
+        }
+
+        // Populate the TWAP message if accumulator will be used
+        if maybe_accumulator_accounts.is_some() {
```

Comment:
I think what Jayant meant here is that you can populate maybe_twap_msg even if the accumulator accounts are not passed. It won't be used but it's ok.